receiver/receivercreator/README.md

```diff
@@ -98,14 +113,16 @@ receivers:
     receivers:
       prometheus_simple:
         # Configure prometheus scraping if standard prometheus annotations are set on the pod.
-        rule: type.pod && annotations["prometheus.io/scrape"] == "true"
+        type: pod
+        rule: annotations["prometheus.io/scrape"] == "true"
```

Comment:
This is the main change in the config, right? I am still not quite sure I understand why the old way was a problem, though. I read the PR description, but still do not understand.
At least from this example config perspective it appears equivalent. Is it that fields like `name` are ambiguous and need to be assigned different values depending on the `type`?

Reply:
So the observer discovers multiple types of endpoints, let's say it discovers:

```
endpoint{type=pod, name=pod1}
endpoint{type=port, pod.name=pod1, port=80}
...
```

We require type be made explicit so that you don't match a pod type and a port type which are incompatible (they have different variables like pod.name vs name). With a pod the user has to manually specify port for the receiver to contact whereas with a port it has the full endpoint to connect to. So it helps keeping users from shooting themselves in the foot with the rules.

So we have a few options for configuring the rule.

1. (Original method) `type.pod && name == "pod1"
2. (Proposed)
```
type: pod
rule: name == "pod1"
3. (Alternative) `type == "pod" && name == "pod1"

Maybe #2 that I've proposed isn't really any better. #1 just always felt hacky because we enforced the user including the type by checking that the rule started with `type.{pod,port}`. But maybe it's worse UX to essentially have the rule split up across 2 config values in yaml (type and rule).

I'll think on it more.